imap_processing/ultra/l1b/ultra_l1b.py

```diff
@@ -0,0 +1,85 @@
+"""Calculates ULTRA L1b."""
+
+import json
+import logging
+import re
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+
+logger = logging.getLogger(__name__)
+
+
+def create_dataset(data_dict, name):
+    """
+    Create xarray for L1b data.
+
+    Parameters
+    ----------
+    data_dict: : dict
+        L1b data dictionary.
+
+    Returns
+    -------
+    dataset : xarray.Dataset
+        xarray.Dataset
+    """
+    # TODO: this will change to actual l1b data dictionary
+    # For now we are using it for retrieving the epoch.
+    dataset = data_dict["imap_ultra_l1a_45de"]
+
+    # Load metadata from the metadata file
+    with open(Path(__file__).parent.parent / "ultra_metadata_example.json") as f:
+        metadata = json.loads(f.read())[name]
+
+    epoch = dataset.coords["epoch"]
+
+    annotated_de_attrs = metadata["dataset_attrs"]
+
+    dataset = xr.Dataset(
+        coords={"epoch": epoch.astype("datetime64[ns]")},
+        attrs=annotated_de_attrs,
+    )
+
+    dataset["x_front"] = xr.DataArray(
+        np.zeros(len(epoch), dtype=np.float32),
+        dims=["epoch"],
+        attrs=metadata["x_front"],
```

Comment:
Filling zeros for data is great. Is `x_front` an actual data variable of annotated DE?

Reply:
Yes. It is the event position on the front foil. L1b-L1c will have zeros. I am almost done with extended events so most of these will be filled soon, but I don't want to risk running out of time.